uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/bookingrequest/BookingRequestActivity.kt

```diff
@@ -228,30 +199,31 @@ class BookingRequestView @JvmOverloads constructor(context: Context,
     }
 
     private fun hideWindow() {
-        if (containerAnimateOut.hasEnded() || !containerAnimateOut.hasStarted()) {
-            presenter.onPaymentFailureDialogCancelled()
-        }
+        presenter.onPaymentFailureDialogCancelled()
+        finish();
     }
 
     override fun onLoadingButtonClick() {
-        if (!isKarhooUser() && bookingRequestPassengerDetailsWidget.findAndfocusFirstInvalid()) {
+        if (!(KarhooUISDKConfigurationProvider.configuration.authenticationMethod() is AuthenticationMethod.KarhooUser)
```

Comment:
the reason why i added the `isKarhooUser()` method is to make the code more readable, we can remove it but it's kind of hard to read like that. Another option is to modify it like this: `KarhooUISDKConfigurationProvider.configuration.authenticationMethod() !is AuthenticationMethod.KarhooUser`

Reply:
Done